internal/component/prometheus/remote/queue/network/manager.go

```diff
@@ -0,0 +1,171 @@
+		metaInbox:   actor.NewMailbox[*types.TimeSeriesBinary](actor.OptCapacity(1)),
+		configInbox: actor.NewMailbox[types.ConnectionConfig](),
+		stats:       seriesStats,
+		cfg:         cc,
+	}
+
+	// start kicks off a number of concurrent connections.
+	var i uint64
+	for ; i < s.connectionCount; i++ {
+		l := newLoop(cc, false, logger, seriesStats)
+		l.self = actor.New(l)
+		s.loops = append(s.loops, l)
+	}
+
+	s.metadata = newLoop(cc, true, logger, metadataStats)
+	s.metadata.self = actor.New(s.metadata)
+	return s, nil
+}
+
+func (s *manager) Start() {
+	s.startLoops()
+	s.configInbox.Start()
+	s.metaInbox.Start()
+	s.inbox.Start()
+	s.self = actor.New(s)
+	s.self.Start()
+}
+
+func (s *manager) SendSeries(ctx context.Context, data *types.TimeSeriesBinary) error {
+	return s.inbox.Send(ctx, data)
+}
+
+func (s *manager) SendMetadata(ctx context.Context, data *types.TimeSeriesBinary) error {
+	return s.metaInbox.Send(ctx, data)
+}
+
+func (s *manager) UpdateConfig(ctx context.Context, cc types.ConnectionConfig) error {
+	return s.configInbox.Send(ctx, cc)
+}
+
+func (s *manager) DoWork(ctx actor.Context) actor.WorkerStatus {
+	// This acts as a priority queue, always check for configuration changes first.
+	select {
+	case cfg, ok := <-s.configInbox.ReceiveC():
+		if !ok {
+			return actor.WorkerEnd
+		}
+		s.updateConfig(cfg)
+		return actor.WorkerContinue
+	default:
+	}
+	select {
+	case <-ctx.Done():
+		s.Stop()
+		return actor.WorkerEnd
+	case ts, ok := <-s.inbox.ReceiveC():
+		if !ok {
+			return actor.WorkerEnd
+		}
+		s.queue(ctx, ts)
+		return actor.WorkerContinue
+	case ts, ok := <-s.metaInbox.ReceiveC():
+		if !ok {
+			return actor.WorkerEnd
+		}
+		err := s.metadata.seriesMbx.Send(ctx, ts)
+		if err != nil {
+			level.Error(s.logger).Log("msg", "failed to send to metadata loop", "err", err)
+		}
+		return actor.WorkerContinue
+	}
+}
+
+func (s *manager) updateConfig(cc types.ConnectionConfig) {
+	// No need to do anything if the configuration is the same.
+	if s.cfg.Equals(cc) {
+		return
+	}
+	s.cfg = cc
+	// TODO @mattdurham make this smarter, at the moment any samples in the loops are lost.
+	// Ideally we would drain the queues and re add them but that is a future need.
+	// In practice this shouldn't change often so data loss should be minimal.
+	// For the moment we will stop all the items and recreate them.
+	s.stopLoops()
+	s.loops = make([]*loop, 0, s.connectionCount)
+	var i uint64
+	for ; i < s.connectionCount; i++ {
+		l := newLoop(cc, false, s.logger, s.stats)
+		l.self = actor.New(l)
+
+		s.loops = append(s.loops, l)
+	}
+
+	s.metadata = newLoop(cc, true, s.logger, s.metaStats)
+	s.metadata.self = actor.New(s.metadata)
+	s.startLoops()
+}
+
+func (s *manager) Stop() {
+	s.stopLoops()
+	s.configInbox.Stop()
+	s.metaInbox.Stop()
+	s.inbox.Stop()
+	s.self.Stop()
+}
+
+func (s *manager) stopLoops() {
+	for _, l := range s.loops {
+		l.stopCalled.Store(true)
```

Comment:
`stopCalled` is set inside `l.Stop()` already - I think we can have just the latter?

Reply:
Good catch!